sdks/java/core/src/main/java/org/apache/beam/sdk/util/state/StateSpec.java

```diff
@@ -36,4 +37,10 @@
    * Use the {@code binder} to create an instance of {@code StateT} appropriate for this address.
    */
   StateT bind(String id, StateBinder<? extends K> binder);
+
+  /**
+   * Given {code coders} are inferred from type arguments defined for this class.
+   * @param coders Array of coders indexed by the type arguments order.
+   */
+  void offerCoders(Coder[] coders);
```

Comment:
That sounds about right. We should document like you have described here. I guess in the end each `StateSpec` implementation can decide what to do on its own. And for my other question, I think document that some entries in the array might be null. My favorite approach is to provide a `Map<TypeVariable, Coder>` but that might be a pain here.